providers/shared/components/ddb/id.ftl

```diff
@@ -0,0 +1,260 @@
+                    {
+                        "Names" : "EncryptionScheme",
+                        "Types" : STRING_TYPE,
+                        "Description" : "A prefix appended to link attributes to show encryption status",
+                        "Default" : ""
+                    }
+                ]
+            },
+            {
+                "Names" : "rootCredential:SecretStore",
+                "Description" : "Use a secret store to manage the root credentials",
+                "Children" : [
+                    {
+                        "Names" : "Username",
+                        "Types" : STRING_TYPE,
+                        "Default" : "root"
+                    },
+                    {
+                        "Names" : "UsernameAttribute",
+                        "Description" : "The attribute of the username to store in the secret",
+                        "Types" : STRING_TYPE,
+                        "Default" : "username"
+                    },
+                    {
+                        "Names" : "PasswordAttribute",
+                        "Description" : "The attribute of the password to store in the secret",
+                        "Types" : STRING_TYPE,
+                        "Default" : "password"
+                    },
+                    {
+                        "Names" : "GenerationRequirements",
+                        "Description" : "When creating the secret using secretstore the policy for generating the secret",
+                        "AttributeSet" : SECRETSTRING_ATTRIBUTESET_TYPE
+                    }
+                    {
+                        "Names" : "Link",
+                        "Description" : "A link to a secret or store that will keep the secret",
+                        "AttributeSet" : LINK_ATTRIBUTESET_TYPE
+                    }
+                ]
+            },
+            {
+                "Names" : "rootCredential:Settings",
+                "Description" : "Store the credentials as settings of the database occurrence",
+                "Children" : [
+                    {
+                        "Names" : "UsernameAttribute",
+                        "Description" : "The setting attribute that contains the username",
+                        "Types" : STRING_TYPE,
+                        "Default" : "MASTER_USERNAME"
+                    },
+                    {
+                        "Names" : "PasswordAttribute",
+                        "Description" : "The setting attribute that contains the password",
+                        "Types" : STRING_TYPE,
+                        "Default" : "MASTER_PASSWORD"
+                    }
+                ]
+            },
+            {
+                "Names" : "Backup",
+                "Children" : [
+                    {
+                        "Names" : "BackupWindow",
+                        "AttributeSet" : BACKUPWINDOW_ATTRIBUTESET_TYPE
+                    },
+                    {
```

Comment:
True but it was very similar to MaintenanceWindow which is used in multiple locations so though to make it an attribute set on the assumption that it would also appear elsewhere (although given the elapsed time - maybe not)

Is there a downside to it being an attribute set?